### PR TITLE
nix-doc: init at 0.3.3

### PR DESCRIPTION
**Pull request by lf-**
###### Motivation for this change

It was requested by @gvolpe :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [???] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [???] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

re: the ??? above, nixpkgs-review did not find anything when I ran it post-commit, and `nix path-info -S` hit the intentionally failing pseudo-package, so I'm not sure how I'm supposed to do that piece.

### CODE REVIEW

**Comment by lf-** [comment]
Apologies about the sneaky edits, there was a pretty nasty bug I found and squished while doing final validation on the package.

**Comment by lf-** [comment]
OK I am done smuggling in changes. It builds and runs now with no issues, and contains two fewer bugs than when I started :)

**Comment by gvolpe** [comment]
@lf- you will be asked to create a separate commit to add yourself as a maintainer as `maintainers: add lf-`. Ideally, this commit should come first :)

**Comment by gvolpe** [comment]
You can also ignore that `???` as it doesn't apply to this new package.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4661,6 +4661,12 @@
       fingerprint = "7FE2 113A A08B 695A C8B8  DDE6 AE53 B4C2 E58E DD45";
     }];
   };
+  lf- = {
```
Comment (IvarWithoutBones) [inline maintainers/maintainer-list.nix:4664]: This should be in a seperate commit.

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, rustPlatform, fetchFromGitHub, pkgs, ... }:
```
Comment (IvarWithoutBones) [inline pkgs/tools/package-management/nix-doc/default.nix:1]: You shouldn't use `pkgs` in derivations, rather import the required packages directly.

```suggestion
{ stdenv, rustPlatform, fetchFromGitHub, pkg-config, boost, nix }:
```

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "lf-";
+    repo = "nix-doc";
+    sha256 = "0vd7159y5w8jjgaw51kfr3z3r50299gvw7vjchpqx3nwmdink8bh";
+  };
+
+  doCheck = true;
+  buildInputs = with pkgs; [ boost nix ];
```
Comment (IvarWithoutBones) [inline pkgs/tools/package-management/nix-doc/default.nix:15]:
```suggestion
  buildInputs = [ boost nix ];
```

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  nativeBuildInputs = with pkgs; [ pkg-config ];
```
Comment (IvarWithoutBones) [inline pkgs/tools/package-management/nix-doc/default.nix:17]: `nativeBuildInputs` are usually sorted above regular `buildInputs`, though I'm not sure if theres any strict guidelines on this.

```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  cargoSha256 = "1xz3qngs8p0s62dq4d46c01z3k1vvgg856767g56b13c38pzfh28";
+
+  meta = with stdenv.lib; {
+    description = "An interactive Nix documentation tool providing a CLI for function search and a Nix plugin for docs in the REPL";
```
Comment (IvarWithoutBones) [inline pkgs/tools/package-management/nix-doc/default.nix:22]:
```suggestion
    description = "An interactive Nix documentation tool";
    longDescription = "An interactive Nix documentation tool providing a CLI for function search and a Nix plugin for docs in the REPL";
```

**Review by IvarWithoutBones — COMMENTED** [review]
Left a few suggestions, other than that LGTM :)

**Comment by lf-** [comment]
Thanks for the review @IvarWithoutBones. The requested changes have been made.

File: pkgs/tools/package-management/nix-doc/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, rustPlatform, fetchFromGitHub, boost, nix, pkg-config, ... }:
```
Comment (IvarWithoutBones) [inline pkgs/tools/package-management/nix-doc/default.nix:1]: The `...` here isnt required.
```suggestion
{ stdenv, rustPlatform, fetchFromGitHub, boost, nix, pkg-config }:
```

Comment (lf-) [reply pkgs/tools/package-management/nix-doc/default.nix]: Changed.

**Comment by lf-** [comment]
@IvarWithoutBones changed :)

**Pull request merged**